### PR TITLE
Unite bless environment variables under `RUSTC_BLESS`

### PR DESCRIPTION
**Pull request by tgross35**
Currently, Clippy and Miri both use an environment variable to indicate that output should be blessed, but they use different variable names. In order to improve consistency, this patch applies the following changes:

- Rename the variable `MIRI_BLESS` (as used in the Miri subtree) to `RUST_BLESS`
- Rename the variable `BLESS` (as used in the Clippy subtree) to `RUST_BLESS`
- Move emitting `RUST_BLESS` into `prepare_cargo_test` so it is always available (I need this for a WIP PR)

---

I prefer something like `RUST_BLESS` to `BLESS` just for a lower chance of conflict (not super common but other tools [do use `BLESS`](https://grep.app/search?q=%22BLESS%22&case=true&words=true&filter[lang][0]=Text&filter[lang][1]=Rust&filter[lang][2]=Python&filter[lang][3]=C%2B%2B&filter[lang][4]=Markdown&filter[lang][5]=C&filter[lang][6]=JSON)), but I can change it to whatever is preferred.

Original discussion: https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/BLESS.20env.20var.3A.20rename.20to.20CLIPPY_BLESS

r? @oli-obk
cc @flip1995

### CODE REVIEW

File: src/tools/miri/test-cargo-miri/run-test.py
Comment (tgross35) [inline src/tools/miri/test-cargo-miri/run-test.py:12]: I think my formatter sorted this by accident... but I guess it's better that way anyway

File: src/bootstrap/test.rs
Comment (flip1995) [inline src/bootstrap/test.rs:2215]: I like that `Gesundheit` bubbled up :smile:

**Review by flip1995 — COMMENTED** [review]
I would name it `RUSTC_BLESS`, if you want a prefix. But then if you work in tool repos like Clippy or MIRI, you'll have to run `RUSTC_BLESS=1 cargo test`, which is a bit weird, because the tool is not rustc. I feel the same with just `RUST_BLESS`. But maybe that's just me :thinking:

File: src/bootstrap/test.rs
Comment (tgross35) [reply src/bootstrap/test.rs:2215]: Aber selbstverständlich 🙂

**Comment by tgross35** [comment]
> I would name it `RUSTC_BLESS`, if you want a prefix. But then if you work in tool repos like Clippy or MIRI, you'll have to run `RUSTC_BLESS=1 cargo test`, which is a bit weird, because the tool is not rustc. I feel the same with just `RUST_BLESS`. But maybe that's just me 🤔

I did have the same thoughts. Any thoughts on `RUSTBLESS` maybe? To match `RUSTFLAGS`/`RUSTDOCFLAGS` (not that those are bootstrap-only...)

Anyhow I set it to `RUSTC_BLESS` for now

**Comment by RalfJung** [comment]
RUSTC_BLESS works for me. We anyway invoke this as "./miri bless" (or hopefully soon "./miri test --bless") so users never see the env var.

**Comment by Manishearth** [comment]
cc @jyn514 as our resident "bootstrap should be consistent dammit" person.

**Comment by jyn514** [comment]
i think the clippy user experience is more important than making bootstrap's code simpler. bootstrap kind of fundamentally has to workaround quirks in the tools it invokes so i place a fairly low priority on removing the workarounds as long as they're reliable.

that said, the inconsistency i see from the *user* perspective is that bootstrap uses `--bless` and clippy uses an env variable. i don't think there's a way to solve that while still allowing clippy to use `cargo test` without a wrapper though, i definitely agree there's a lot of value in having `cargo test` work out of the box.

so my preference is probably to close this PR and only change env variables within rust-lang/rust itself to be consistent, not in any subtrees.

**Comment by flip1995** [comment]
> i think the clippy user experience is more important than making bootstrap's code simpler.

We're not married to those env vars, we just added them like 1 week ago. So changing them doesn't cause any harm. And I myself have no objective objections against `RUSTC_BLESS` (or similar), just that it subjectively _feels_ weird to me. So don't block this PR on this.





> that said, the inconsistency i see from the _user_ perspective is that bootstrap uses `--bless` and clippy uses an env variable

That's what MIRI also does. But now in miri the env var is called `MIRI_BLESS` and in Clippy it's just called `BLESS`. The reason for this is how `ui-test` (not `compiletest-rs`!) works. In the Clippy repo itself we have a `cargo bless` alias. But not sure if that would work in bootstrap?

**Comment by RalfJung** [comment]
From a user perspective, Miri doesn't use an env var, it uses `./miri bless`. The fact that there exists an env var is an internal implementation detail, not visible to users.

**Comment by tgross35** [comment]
I think at least the portion of this it at sets a commonly usable variable should be fine and needed (at least I use it in #112697). Clippy and Miri could always split off their own variables again if it ever turns out they need finer control - but it seems like there isn’t much objection to trying to use the same one as-is? At least until there is something better than the env solution, which sounds cleaner but also not necessarily close.

I’ll make it your call, I can adjust as needed

r? @jyn514

**Comment by jyn514** [comment]
i don't have time for reviews, sorry

r? bootstrap

**Comment by blyxyas** [comment]
> BLESS just for a lower chance of conflict (not super common but other tools [do use BLESS](https://grep.app/search?q=%22BLESS%22&case=true&words=true&filter%5Blang%5D%5B0%5D=Text&filter%5Blang%5D%5B1%5D=Rust&filter%5Blang%5D%5B2%5D=Python&filter%5Blang%5D%5B3%5D=C%2B%2B&filter%5Blang%5D%5B4%5D=Markdown&filter%5Blang%5D%5B5%5D=C&filter%5Blang%5D%5B6%5D=JSON))

AFAIK, BLESS' intended use is as `$ BLESS=true my_command ...`. So it gets removed as soon as `my_command` is over. In this case, collisions wouldn't be an issue (I hope no one is working on blessing a Rust test and a Facebook test at the same time :sweat_smile: )

**Comment by clubby789** [comment]
I don't have a strong opinion on the variable name to use

r? bootstrap

**Comment by oli-obk** [comment]
rustfmt now also has an env var for blessing: https://github.com/rust-lang/rust/pull/114054

**Review by oli-obk — APPROVED** [review]
lgtm, and as you said, we can bikeshed in the future.

needs a rebase and rustfmt taken into the fold

**Comment by tgross35** [comment]
@oli-obk thanks for taking the potato, this should be in sync again. I noted in your linked PR that you check the variable is not set to `0`, so I made that consistent everywhere.

> Some changes occurred in src/tools/cargo
> 
> cc @ehuss

This was a submodule whoops, should be gone now

**Pull request merged**
